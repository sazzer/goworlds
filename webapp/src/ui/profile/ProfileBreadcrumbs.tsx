import React, {FunctionComponent} from 'react';
import {Breadcrumb} from "semantic-ui-react";
import {Link} from "react-router-dom";
import {Trans, useTranslation} from "react-i18next";
import {NamedUser} from "./NamedUser";

/** The type for the breadcrumbs */
export declare type ProfileBreadcrumbsProps = {
    user: NamedUser
};

/**
 * Breadcrumbs to render on the Profile page
 * @constructor
 */
export const ProfileBreadcrumbs: FunctionComponent<ProfileBreadcrumbsProps> = ({user}) => {
    const { t } = useTranslation();

    return (
        <Breadcrumb>
            <Breadcrumb.Section>
                <Link to="/">{t('profile.breadcrumbs.home')}</Link>
            </Breadcrumb.Section>
            <Breadcrumb.Divider />
            <Breadcrumb.Section active>
                <Trans i18nKey='profile.breadcrumbs.profile' values={user}>
                    Profile: <Link to="/profile">{user.name}</Link>
                </Trans>
            </Breadcrumb.Section>
        </Breadcrumb>
    );
}
