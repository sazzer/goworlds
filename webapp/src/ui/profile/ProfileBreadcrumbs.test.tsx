import React from 'react';
import {ProfileBreadcrumbs} from './ProfileBreadcrumbs';
import {render} from '@testing-library/react';
import {MemoryRouter} from "react-router";

jest.mock('../common/Authenticated', () => {
    const Authenticated = jest.fn(() => 'Mock Authenticated');
    return {
        Authenticated,
    };
});

/** Set up the component to test */
function setup(name: string = 'Username') {
    return {
        element: render(<MemoryRouter><ProfileBreadcrumbs user={{name}} /></MemoryRouter>)
    };
}

it('renders without crashing', () => {
    const {element} = setup();

    expect(element.baseElement).toMatchSnapshot();
});
