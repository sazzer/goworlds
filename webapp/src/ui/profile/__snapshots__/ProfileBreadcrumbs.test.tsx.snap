// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<body>
  <div>
    <div
      class="ui breadcrumb"
    >
      <div
        class="section"
      >
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="divider"
      >
        /
      </div>
      <div
        class="active section"
      >
        Profile: 
        <a
          href="/profile"
        >
          Username
        </a>
      </div>
    </div>
  </div>
</body>
`;
